### DISCUSSION TITLE
nextjs + graphql-codegen is not working in `ssr` and `ssg`

### DISCUSSION THREAD

**Discussion by amantiwari1** [dotansimha/graphql-code-generator#6070] [Q&A]
I can't find a solution in google, youtube, etc none of them are working so please if you think it is working so please send your repo or codesandbox as possible thank you in advance

**Comment by Urigo** [comment]
Hi @amantiwari1 , can you share an example of what you are doing and what is not working?
thank you

**Reply by amantiwari1** [reply]
It is working fine in client side like react as same as nextjs
But I wanted to ssg and ssr but I can't find example of this nextjs

**Reply by amantiwari1** [reply]
Do you have example of ssg and ssr in nextjs please share

**Reply by amantiwari1** [reply]
@Urigo

**Reply by Urigo** [reply]
I don't have one ready, maybe you can grab one from Next's repo's examples, add codegen and share here?

**Reply by Urigo** [reply]
here are some things to check out:

1. https://www.graphql-code-generator.com/docs/plugins/typescript-apollo-next
2. https://github.com/correttojs/graphql-codegen-apollo-next-ssr
3. https://github.com/vercel/next.js/tree/canary/examples/with-typescript-graphql

**Reply by amantiwari1** [reply]
All these link example are not enough info and old  class components and there is no ssg please can you  make example combine ssr and ssg and share code please @Urigo

**Reply by Urigo** [reply]
I won't make an example.  If you want, you can take these as references, try to make one yourself and I can help you fix it and share it with the community

**Reply by amantiwari1** [reply]
@Urigo https://codesandbox.io/s/sparkling-glade-y0p9f

**Comment by dotansimha** [comment]
I don't see any details here for reproducing or fixing. Please provide additional information about the error, codegen version, nextjs version, what client you are using, and what plugins you are using.

**Reply by amantiwari1** [reply]
I am sorry, I am super busy , the main problem is I tried using hook In codegen in functional comments then results is it is not rendered html instead they used fetch api from client side, the problem is I can't find solution how to fix that , I would great if you can help me

**Reply by amantiwari1** [reply]
I tried hook in staticProps  function , it results hook are not supported not this function which are strictly rules

**Reply by amantiwari1** [reply]
@dotansimha  https://codesandbox.io/s/sparkling-glade-y0p9f

**Comment by amantiwari1** [comment]
here is codestandbox https://codesandbox.io/s/sparkling-glade-y0p9f

**Reply by dotansimha** [reply]
I see, thank you. 
The issue here has nothing to do with codegen. The plugins you are using there are wrapping Apollo-Client, and just add types to it. 
For more information about SSR/NextJS integration with Apollo-Client, you can use this: https://www.apollographql.com/docs/react/performance/server-side-rendering/

**Reply by amantiwari1** [reply]
My question is how do I get data in  function (it doesn't work  for hook  ) without jsx,  hook and class components?

**Reply by dotansimha** [reply]
Maybe this can help: https://github.com/dotansimha/graphql-code-generator/discussions/6138#discussioncomment-896899
